### PR TITLE
adding new Query Condition constructor that replaces Class with tiledb.Datatype

### PR DESCRIPTION
**Pull request by DimitrisStaratzis**
API improvement. Previous implementation could lead user to not be able to fully control the datatype in the Query Condition as they would pass a Class object which was then internally translated to a TileDB datatype. This unnecessary middle step is now removed. 

Tests have also been modified to use the new constructor.

### CODE REVIEW

**Review by ihnorton — CHANGES_REQUESTED** [review]
We need to deprecate the old signature if possible, so that this is not a breaking change.

File: src/main/java/io/tiledb/java/api/QueryCondition.java [hunk 1]
```diff
@@ -84,6 +86,58 @@ public QueryCondition(
...
+        array = new NativeArray(ctx, value, type.javaClass());
+      } else {
+        if (type != null) {
+          int byteSize = type.getNativeSize();
+          array = new NativeArray(ctx, byteSize, type);
+          array.setItem(0, value);
+        }
+      }
+      conditionp = tiledb.tiledb_query_condition_tpp_value(conditionpp);
+      if (type == null) {
```
Comment (ihnorton) [inline src/main/java/io/tiledb/java/api/QueryCondition.java:120]: The logic is a little bit hard to follow here. It seems like you could simplify quite a bit by moving this check to the top?

Comment (DimitrisStaratzis) [reply src/main/java/io/tiledb/java/api/QueryCondition.java]: I have simplified the new constructor

**Review by ihnorton — APPROVED** [review]
LGTM, thanks for the updates.
